src/PausableRole.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity ^0.8.24;
+
+import {IPausableRole} from "./interfaces/IPausableRole.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Pausable} from "@openzeppelin/contracts/security/Pausable.sol";
+
+/// @notice Allow owner and multiple accounts to pause but only owner can unpause
+/// @dev Potentially allow security partners to programatically pause()
+abstract contract PausableRole is IPausableRole, Ownable, Pausable {
+    mapping(address => bool) public hasPausableRole;
+
+    modifier onlyPausableRoleOrOwner() {
+        if (msg.sender != owner() && hasPausableRole[msg.sender] == false) revert NoPausableRole();
```

Comment:
`hasPausableRole[msg.sender] == false` can be replaced with `!hasPausableRole[msg.sender]`